### PR TITLE
support for interactions

### PR DESCRIPTION
**Pull request by phspo**
added support for interactions:

(
a interaction is a contrast that fixes several variables:

assume there are three dimensions: A, B and C: for each dimension there might be multiple conditions possible so for instance A can assume either 1 or 2, the same goes for B and C. so far we can use the workflow to inspect the differences between A1 and A2, it might also be interested to see how A1 and A2 differ specifically in the environment of B1 and C2 (or only B1 and either C1 or C2).

In other words: We might have a complex experimental design matrix and want to be able to make any comparison.
)

I also tried adding a little toy example to help intuitively understand the concept but due to the latest changes (adding qc) the synthetic data was deemed to unrealistic and fails the qc.

[exampledata.zip](https://github.com/snakemake-workflows/rna-seq-star-deseq2/files/3248265/exampledata.zip)

### CODE REVIEW

**Comment by phspo** [comment]
sounds good, although this is a question of how complex the interface to deseq2 should be for the user. i think one nice advantage of this workflow as opposed to just accessing deseq2 directly via R is that for simple applications one does not need to mess with the formula at all. how would you envision a free design formula would look like in terms of snakemake, what would be configurable in the config file etc.?

**Comment by johanneskoester** [comment]
Sorry for the delay! Similar to this in the Kallisto pipeline: https://github.com/snakemake-workflows/rna-seq-kallisto-sleuth/blob/master/config.yaml#L21
